transport/src/main/java/io/netty/channel/DefaultChannelConfig.java

```diff
@@ -374,11 +383,22 @@ public ChannelConfig setWriteBufferLowWaterMark(int writeBufferLowWaterMark) {
             throw new IllegalArgumentException(
                     "writeBufferLowWaterMark must be >= 0");
         }
-        this.writeBufferLowWaterMark = writeBufferLowWaterMark;
+        this.writeBufferWaterMark.setWriteBufferLowWaterMark(writeBufferLowWaterMark);
         return this;
     }
 
     @Override
+    public ChannelConfig setWriteBufferWaterMark(WriteBufferWaterMark writeBufferWaterMark) {
```

Comment:
You also need to ensure you override these method in all sub-classes and return the correct type for proper method chaining.